test/functional/apps/discover/_discover.js

```diff
@@ -0,0 +1,134 @@
+          common.debug('clickTimepicker');
+          return discoverPage.clickTimepicker();
+        })
+        .then(function () {
+          common.debug('setAbsoluteRange');
+          return discoverPage.setAbsoluteRange(fromTime, toTime);
+        })
+        .then(function () {
+          common.debug('collapseTimepicker');
+          return discoverPage.collapseTimepicker();
+        });
+      });
+
+
+      bdd.describe('query', function () {
+        var queryName1 = 'Query # 1';
+
+        bdd.it('should show correct time range string', function () {
+          var expectedTimeRangeString =
+          'September 19th 2015, 06:31:44.000 to September 23rd 2015, 18:31:44.000';
+          return discoverPage.getTimespanText()
+          .then(function (actualTimeString) {
+            expect(actualTimeString).to.be(expectedTimeRangeString);
+          })
+          .catch(common.handleError(this));
+        });
+
+        bdd.it('save query should show toast message and display query name', function () {
+          var expectedSavedQueryMessage = 'Discover: Saved Data Source "' + queryName1 + '"';
+          this.timeout = 60000;
+          return discoverPage.saveSearch(queryName1)
+          .then(function () {
+            return discoverPage.getToastMessage(2000);
+          })
+          .then(function (toastMessage) {
+            expect(toastMessage).to.be(expectedSavedQueryMessage);
+          })
+          .then(function () {
+            return discoverPage.waitForToastMessageGone();
+          })
+          .then(function () {
+            return discoverPage.getCurrentQueryName();
+          })
+          .then(function (actualQueryNameString) {
+            expect(actualQueryNameString).to.be(queryName1);
+          })
+          .catch(common.handleError(this));
+        });
+
+        bdd.it('load query should show query name', function () {
+          return discoverPage.loadSavedSearch(queryName1)
+          .then(function () {
+            return common.tryForTime(15000, function () {
+              return discoverPage.getCurrentQueryName()
+              .then(function (actualQueryNameString) {
+                expect(actualQueryNameString).to.be(queryName1);
+              });
+            });
+          })
+          .catch(common.handleError(this));
+        });
+
+        bdd.it('should show the correct bar chart', function () {
+          var expectedBarChartData = [0,0,0,0,1.0968749999999972,7.6781250000000085,
+            37.87875,92.210625,108.590625,71.80875,23.54625,4.753124999999997,
+            2.1206249999999898,7.60499999999999,35.319374999999994,85.044375,110.199375,
+            70.05375000000001,23.180625000000006,4.0218750000000085,1.2431250000000063,
+            6.435000000000002,36.416250000000005,88.408125,108.81,
+            69.395625,22.522499999999994,5.4112499999999955,0.29249999999998977,0,0,0,0,0,0,0,0];
+          this.timeout = 80000;
+          return common.sleep(4000)
```

Comment:
Similar to comment above, I increased these timeouts a lot because tests that pass over 100 times in a row on my laptop fail on Travis and it's rather hard to figure out the exact point of failure without a lot of logging.  I could shorten these timeouts and sleeps, but what works locally can still fail on Travis.  So I have to trial and error it.
The error caused when getBarChartData runs before the chart is ready is seen in this build;
https://travis-ci.org/elastic/kibana/builds/90999280

Error: JavaScriptError: [POST http://localhost:4444/wd/hub/session/8ea29788-4142-4916-982c-f1021d81502f/execute / {"script":"return arguments[0].getAttribute(arguments[1]);","args":[{"ELEMENT":"246"},"fill"]}] arguments[0].getAttribute is not a function